src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -849,20 +906,20 @@ static int rbd_nbd(int argc, const char *argv[])
     } else if (ceph_argparse_witharg(args, i, &nbds_max, err, "--nbds_max", (char *)NULL)) {
       if (!err.str().empty()) {
         cerr << err.str() << std::endl;
```

Comment:
I still think it is a good idea to suppress error message when parsing cmdline. It could be a boolean param to `parse_args` to tell if it needs to print an error message, or `std::ostream &err`, so the message will be stored there, and you can print it or ignore after the function returns.